cleverpush/src/main/java/com/cleverpush/CleverPush.java

```diff
@@ -1530,127 +1523,36 @@ public void getAvailableTopics(ChannelTopicsListener listener) {
         return topics;
     }
 
-    public void addSubscriptionTags(String[] tagIds) {
-        currentPositionOfTagToAdd = 0;
-        addSubscriptionTag(tagIds[currentPositionOfTagToAdd], new OnAddTagCompleted() {
-            @Override
-            public void onAddTagCompleted() {
-                if (currentPositionOfTagToAdd != tagIds.length - 1) {
-                    currentPositionOfTagToAdd++;
-                    addSubscriptionTag(tagIds[currentPositionOfTagToAdd], this::onAddTagCompleted);
-                }
-            }
-        });
+    public void addMultipleSubscriptionTags(String[] tagIds) {
+        this.waitForTrackingConsent(() -> new Thread(() -> this.getSubscriptionId(subscriptionId -> {
```

Comment:
duplicate code, only difference is addMultipleSubscriptionTagsHelper.addMultipleSubscriptionTags();
and
addMultipleSubscriptionTagsHelper.addSubscriptionTag();

so try exporting everything around

```
{
            AddSubscriptionTags addMultipleSubscriptionTagsHelper = new AddSubscriptionTags(subscriptionId,this.channelId,tagId);
            addMultipleSubscriptionTagsHelper.addSubscriptionTag();
        })).start())

```